File: Logica C/log3.c
#include <stdio.h>

int main(void) {
  int valor;
  int i = 1;

  while (1) { //enquanto for verdadeiro, executa
    printf("Informe um valor inteiro p/ prosseguir ou um negativo p/ parar: ");
    scanf("%d", &valor);

    if (valor < 0) {
      break; //para o for quando o valor for 0
    }

    if (valor % 2 == 0) {
      printf("-> É um número par.\n");
    }

    i++; //executa novamente a estrutura de repetição
  }

  return 0;
}

File: Logica JS/log3.js
let valor;
let i = 1;
let outputElement = document.getElementById('output');

while (true) {

    valor = prompt("Informe um valor inteiro p/ prosseguir ou um negativo para parar:");

    valor = parseInt(valor);

    if (valor < 0) {
        outputElement.innerHTML += "-> Finalizando...<br>";
        break;
    }

    if (valor % 2 === 0) {
        outputElement.innerHTML += "-> É um número par.<br>";
    }
    else{
        outputElement.innerHTML += "-> É um número ímpar.<br>";
    }

    i++;
}